### DISCUSSION TITLE
Dose React Route support name based routing ?

### DISCUSSION THREAD

**Discussion by Alceatraz** [remix-run/react-router#11995] [Q&A]
Can ReactRoute redirect to a name based URL ?

For example, This is a actual working demo in Vue with Vue Route. The way I'm using of hash is pretty hacking. But is there any apporch in React Route ?

```js
import {createRouter, createWebHistory} from 'vue-router';

export router = createRouter({

    history: createWebHistory(),
    routes: [
        {
            name: 'home',
            path: '/',
            component: HomePage,
            beforeEnter: async (to, from) => {
                if (to.hash === '') return;
                return {name: to.hash.substring(1)};
            }
        },
        {
            name: 'setting',
            path: '/#setting',
            component: SettingPage,
        },
        {
            name: 'about',
            path: '/#about',
            component: AboutPage,
        },
    ]
});
```

**Comment by pingusama1** [comment]
I have create this discussion for that: https://github.com/remix-run/react-router/discussions/11951

**Reply by Alceatraz** [reply]
> I have create this discussion for that: #11951

Are you still continue on this? They not accept direct PR.

**Reply by pingusama1** [reply]
I haven't continue to working since the PR reject but I can continue this work. I can, for example, create unit test for this feature.

**Reply by pingusama1** [reply]
But I think my job could be a base for this feature

**Reply by pingusama1** [reply]
I have write unit tests but it failed and I can't undestand why

**Comment by pingusama1** [comment]
[useNamedRoutes-test.txt](https://github.com/user-attachments/files/17115796/useNamedRoutes-test.txt)

**Reply by pingusama1** [reply]
I have this error : useNamedRoutes must be used within a data router.
